apps/react/src/components/InputDate.tsx

```diff
@@ -1,117 +1,66 @@
+  
+    const handleChange = useCallback((e) => {
+        const newDateValue = e.target.value; // "YYYY-MM-DD" format
+        setInputValue(newDateValue);
+        if (newDateValue.match(/^\d{4}-\d{2}-\d{2}$/)) {
+          onChange(stringToDate(newDateValue.replace(/-/g, "_"))); // Convert to "DD_MM_YYYY" format if needed
+        }
+      }, [onChange]);
+  
 
-const InputDate: React.FC<InputDate_Input> = ({
-  options,
-  value,
-  onChange,
-  ...props
-}) => {
-  const [show, setShow] = useState(false);
-  const datePickerWrapperRef = useRef(null);
 
-  const onChangeCallback = useCallback(
-    (date) => {
-      const dateString = DateTime.fromJSDate(date).toISODate();
-      setInputValue(dateString);
-      onChange?.(date);
-      // Keep the calendar open even after selecting a date
-    },
-    [onChange],
-  );
-
-  const opts = useMemo(
-    () => ({
-      ...options,
-      theme: {
-        selected: "bg-blue-500 hover:bg-blue-700 text-white",
-      },
-    }),
-    [options],
-  );
-
-  const [inputValue, setInputValue] = useState(
-    DateTime.fromJSDate(value).toISODate(),
-  );
-
-  return (
-    <>
-      <div className="relative">
-        <button
-          onClick={() => setShow(!show)}
-          className="absolute right-0 top-0 mr-3 mt-2"
-        >
-          <FaRegCalendarAlt className="text-xl" /> {/* Calendar icon */}
-        </button>
-        <input
-          type="text"
-          value={inputValue}
-          onChange={(e) => {
-            setInputValue(e.target.value);
-          }}
-          onBlur={(e) => {
-            // Check whether in correct format
-            const pattern: RegExp = /^\d{4}-\d{2}-\d{2}$/;
-            if (!pattern.test(inputValue)) {
-              return;
-            }
-            // Check whether it's a valid date
-            if (Number.isNaN(Date.parse(inputValue))) {
-              return;
-            }
-
-            const [year, month, day] = inputValue.split("-");
-
-            onChangeCallback(stringToDate(`${day}_${month}_${year}`));
-          }}
-          className={COMMON_INPUT_CLASSES + " w-full"}
-          placeholder="YYYY-MM-DD"
-        />
-        {show && (
-          <div className="date-picker-wrapper" ref={datePickerWrapperRef}>
-            <Datepicker
-              {...props}
-              show={show}
-              setShow={setShow}
-              onChange={onChangeCallback}
-              options={opts}
+      return (
+        <label style={{ display: 'block', marginBottom: '10px', fontSize: '16px' }}>
+          <div style={{ position: 'relative' }}>
+            <input
+              type="date"
+              name="coverageStartDate"
+              value={inputValue}
+              onChange={handleChange}
+              onBlur={onBlur}
+              className={COMMON_INPUT_CLASSES + " pl-4 pr-2 py-2 border rounded"}
+              style={{ display: 'block', width: '100%', fontSize: '16px', lineHeight: '20px' }}
```

Comment:
I would advise against initializing objects directly in the properties. To determine if a component should be re-rendered, React compares what was inputted as a property against the old value for equality using the Object.is() comparison function. Since objects are passed by reference in JS, the "old" { display: 'block', width: '100%', fontSize: '16px', lineHeight: '20px' } will never equal the "new" { display: 'block', width: '100%', fontSize: '16px', lineHeight: '20px' }.